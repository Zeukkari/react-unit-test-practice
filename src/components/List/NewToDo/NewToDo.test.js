import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import NewToDo from './NewToDo';

describe('NewToDo', () => {
  it('renders', () => {
    const newToDo = shallow(<NewToDo submit={() => {}} />);

    expect(toJson(newToDo)).toMatchSnapshot();
  });

  it('updates state on input change', () => {
    const newToDo = shallow(<NewToDo submit={() => {}} />);

    newToDo.find('.new-to-do__input').simulate('change', {
      target: { value: 'Give a talk on unit testing!' },
    });

    expect(newToDo.state().toDo).toBe('Give a talk on unit testing!');
  });

  it('calls props.submit on submit', () => {
    const submit = jest.fn();
    const newToDo = shallow(<NewToDo submit={submit} />);

    newToDo
      .find('.new-to-do__input')
      .simulate('change', { target: { value: 'Hit 100% coverage' } });
    newToDo.find('.new-to-do').simulate('submit', { preventDefault() {} });

    expect(submit).toHaveBeenCalledWith('Hit 100% coverage');
  });

  it('clears state on submit', () => {
    const newToDo = shallow(<NewToDo submit={() => {}} />);

    newToDo
      .find('.new-to-do__input')
      .simulate('change', { target: { value: 'Hit 100% coverage' } });
    newToDo.find('.new-to-do').simulate('submit', { preventDefault() {} });

    expect(newToDo.state().toDo).toBe('');
  });

  it('calls preventDefault on submit', () => {
    const preventDefault = jest.fn();
    const newToDo = shallow(<NewToDo submit={() => {}} />);

    newToDo
      .find('.new-to-do__input')
      .simulate('change', { target: { value: 'Hit 100% coverage' } });
    newToDo.find('.new-to-do').simulate('submit', { preventDefault });

    expect(preventDefault).toHaveBeenCalled();
  });
});
