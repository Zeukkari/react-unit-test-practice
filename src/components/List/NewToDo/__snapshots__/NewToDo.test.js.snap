// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewToDo renders 1`] = `
<form
  className="new-to-do"
  onSubmit={[Function]}
>
  <input
    className="new-to-do__input"
    onChange={[Function]}
    placeholder="What needs to be done?"
    type="text"
    value=""
  />
</form>
`;
